File: test_emoji.java
public class test_emoji {
    public static void main(String[] args) {
        String smiley = "\uD83D\uDE00";
        for (int i = 0; i < smiley.length(); i++) {
            int codeUnit = (int)smiley.charAt(i);
            System.out.println(String.format("%06x", codeUnit));
        }
    }
}

File: test_emoji.kt
fun main() {
    val smiley = "\uD83D\uDE00"
    println("Smiley length: ${smiley.length}")
    for (i in 0 until smiley.length) {
        val codeUnit = smiley[i].code
        println(String.format("%%06x", codeUnit))
    }
}
